From 66fcde4ae4a52ae3edb1cf237ce2c22d08d7a062 Mon Sep 17 00:00:00 2001
From: Rich Felker <dalias@aerifal.cx>
Date: Wed, 11 Jun 2014 23:38:44 -0400
Subject: [PATCH] support optional-argument extension to getopt via
 double-colon

this extension is not incompatible with the standard behavior of the
function, not expensive, and avoids requiring a replacement getopt
with full GNU extensions for a few important apps including busybox's
sed with the -i option.
---
 src/misc/getopt.c | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/src/misc/getopt.c b/src/misc/getopt.c
index f1a1639..8a2e4d5 100644
--- a/src/misc/getopt.c
+++ b/src/misc/getopt.c
@@ -65,8 +65,11 @@ int getopt(int argc, char * const argv[], const char *optstring)
 			}
 			return '?';
 		}
-		optarg = argv[optind++] + optpos;
-		optpos = 0;
+		if (optstring[i+2] == ':') optarg = 0;
+		if (optstring[i+2] != ':' || optpos) {
+			optarg = argv[optind++] + optpos;
+			optpos = 0;
+		}
 	}
 	return c;
 }
-- 
2.0.0

